shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1327,11 +1327,36 @@ content::WebContents* WebContents::OpenURLFromTab(
 void WebContents::BeforeUnloadFired(content::WebContents* tab,
                                     bool proceed,
                                     bool* proceed_to_fire_unload) {
+  LOG(INFO) << "BeforeUnloadFired: proceed=" << proceed;
   // Note that Chromium does not emit this for navigations.
 
   // Emit returns true if preventDefault() was called, so !Emit will be true if
   // the event should proceed.
   *proceed_to_fire_unload = !Emit("-before-unload-fired", proceed);
+
+  // If we were deferring a navigation, either proceed with it or emit a
+  // did-fail-load for the aborted attempt.
+  if (waiting_for_beforeunload_) {
```

Comment:
Is it on purpose that these changes are also part of this PR? I assume they slipped in from working on a different issue.